### PR TITLE
Fixed lambdify usage in stats

### LINKED ISSUE #20292: Test failures from stats and lambdify in master
The stats tests fail on current master. Possibly this depends on the numpy version.

This is a release blocker for 1.7.

```console
$ bin/test sympy/stats/tests/test_continuous_rv.py
============================================================= test process starts =============================================================
executable:         /Users/enojb/current/sympy/38venv/bin/python  (3.8.1-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       gmpy 2.0.8
numpy:              1.18.3
random seed:        893412
hash randomization: on (PYTHONHASHSEED=2425498222)

sympy/stats/tests/test_continuous_rv.py[88] ...w....E...................w............................f.....E..........ww.........Es.     [FAIL]

________________________________________________________________ slowest tests ________________________________________________________________
sympy/stats/tests/test_continuous_rv.py::test_trapezoidal - Took 14.720 seconds
sympy/stats/tests/test_continuous_rv.py::test_exponential - Took 22.614 seconds
sympy/stats/tests/test_continuous_rv.py::test_prefab_sampling - Took 26.954 seconds
_______________________________________________________________________________________________________________________________________________
_______________________________________ sympy/stats/tests/test_continuous_rv.py:test_sample_continuous ________________________________________
Traceback (most recent call last):
  File "/Users/enojb/current/sympy/sympy/sympy/stats/tests/test_continuous_rv.py", line 332, in test_sample_continuous
    assert next(sample(Z)) in Z.pspace.domain.set
  File "/Users/enojb/current/sympy/sympy/sympy/stats/rv.py", line 1221, in return_generator
    d = ps.sample(size=size, library=library)  # a dictionary that maps RVs to values
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 601, in sample
    return {self.value: self.distribution.sample(size, library=library)}
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 290, in sample
    samps = _get_sample_class_crv[library](self, size)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 151, in __new__
    return cls._sample_scipy(dist, size)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 165, in _sample_scipy
    return scipy_rv.rvs(size=size)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 980, in rvs
    vals = self._rvs(*args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 913, in _rvs
    Y = self._ppf(U, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 928, in _ppf
    return self._ppfvec(q, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1673, in _ppf_single
    while self._ppf_to_solve(right, q, *args) < 0.:
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1654, in _ppf_to_solve
    return self.cdf(*(x, )+args)-q
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1834, in cdf
    place(output, cond, self._cdf(*goodargs))
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1709, in _cdf
    return self._cdfvec(x, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1706, in _cdf_single
    return integrate.quad(self._pdf, _a, x, args=args)[0]
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/integrate/quadpack.py", line 341, in quad
    retval = _quad(func, a, b, args, full_output, epsabs, epsrel, limit,
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/integrate/quadpack.py", line 453, in _quad
    return _quadpack._qagse(func,a,b,args,full_output,epsabs,epsrel,limit)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 162, in _pdf
    return handmade_pdf(x)
  File "<lambdifygenerated-2>", line 2, in _lambdifygenerated
    return (select([logical_and.reduce((greater_equal(Dummy_1448, 0),less(Dummy_1448, PINF))),True], [exp(-Dummy_1448),0], default=nan))
AttributeError: 'function' object has no attribute 'reduce'
_______________________________________________________________________________________________________________________________________________
________________________________________ sympy/stats/tests/test_continuous_rv.py:test_prefab_sampling _________________________________________
Traceback (most recent call last):
  File "/Users/enojb/current/sympy/sympy/sympy/stats/tests/test_continuous_rv.py", line 1332, in test_prefab_sampling
    assert next(sample(var)) in var.pspace.domain.set
  File "/Users/enojb/current/sympy/sympy/sympy/stats/rv.py", line 1221, in return_generator
    d = ps.sample(size=size, library=library)  # a dictionary that maps RVs to values
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 601, in sample
    return {self.value: self.distribution.sample(size, library=library)}
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 290, in sample
    samps = _get_sample_class_crv[library](self, size)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 151, in __new__
    return cls._sample_scipy(dist, size)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 165, in _sample_scipy
    return scipy_rv.rvs(size=size)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 980, in rvs
    vals = self._rvs(*args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 913, in _rvs
    Y = self._ppf(U, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 928, in _ppf
    return self._ppfvec(q, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1678, in _ppf_single
    return optimize.brentq(self._ppf_to_solve,
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/optimize/zeros.py", line 780, in brentq
    r = _zeros._brentq(f, a, b, xtol, rtol, maxiter, args, full_output, disp)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1654, in _ppf_to_solve
    return self.cdf(*(x, )+args)-q
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1834, in cdf
    place(output, cond, self._cdf(*goodargs))
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1709, in _cdf
    return self._cdfvec(x, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1706, in _cdf_single
    return integrate.quad(self._pdf, _a, x, args=args)[0]
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/integrate/quadpack.py", line 341, in quad
    retval = _quad(func, a, b, args, full_output, epsabs, epsrel, limit,
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/integrate/quadpack.py", line 453, in _quad
    return _quadpack._qagse(func,a,b,args,full_output,epsabs,epsrel,limit)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 162, in _pdf
    return handmade_pdf(x)
  File "<lambdifygenerated-216>", line 2, in _lambdifygenerated
    return (select([logical_and.reduce((greater_equal(Dummy_12577, 0),less_equal(Dummy_12577, 1))),True], [1,0], default=nan))
AttributeError: 'function' object has no attribute 'reduce'
_______________________________________________________________________________________________________________________________________________
__________________________________________ sympy/stats/tests/test_continuous_rv.py:test_sample_scipy __________________________________________
Traceback (most recent call last):
  File "/Users/enojb/current/sympy/sympy/sympy/stats/tests/test_continuous_rv.py", line 1633, in test_sample_scipy
    samps = next(sample(X, size=size, library='scipy'))
  File "/Users/enojb/current/sympy/sympy/sympy/stats/rv.py", line 1221, in return_generator
    d = ps.sample(size=size, library=library)  # a dictionary that maps RVs to values
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 601, in sample
    return {self.value: self.distribution.sample(size, library=library)}
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 290, in sample
    samps = _get_sample_class_crv[library](self, size)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 151, in __new__
    return cls._sample_scipy(dist, size)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 165, in _sample_scipy
    return scipy_rv.rvs(size=size)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 980, in rvs
    vals = self._rvs(*args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 913, in _rvs
    Y = self._ppf(U, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 928, in _ppf
    return self._ppfvec(q, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1678, in _ppf_single
    return optimize.brentq(self._ppf_to_solve,
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/optimize/zeros.py", line 780, in brentq
    r = _zeros._brentq(f, a, b, xtol, rtol, maxiter, args, full_output, disp)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1654, in _ppf_to_solve
    return self.cdf(*(x, )+args)-q
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1834, in cdf
    place(output, cond, self._cdf(*goodargs))
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1709, in _cdf
    return self._cdfvec(x, *args)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2091, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2167, in _vectorize_call
    outputs = ufunc(*inputs)
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 1706, in _cdf_single
    return integrate.quad(self._pdf, _a, x, args=args)[0]
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/integrate/quadpack.py", line 341, in quad
    retval = _quad(func, a, b, args, full_output, epsabs, epsrel, limit,
  File "/Users/enojb/current/sympy/38venv/lib/python3.8/site-packages/scipy/integrate/quadpack.py", line 453, in _quad
    return _quadpack._qagse(func,a,b,args,full_output,epsabs,epsrel,limit)
  File "/Users/enojb/current/sympy/sympy/sympy/stats/crv.py", line 162, in _pdf
    return handmade_pdf(x)
  File "<lambdifygenerated-285>", line 2, in _lambdifygenerated
    return (select([logical_and.reduce((greater_equal(Dummy_15389, 0),less_equal(Dummy_15389, 1))),True], [1,0], default=nan))
AttributeError: 'function' object has no attribute 'reduce'

========================== tests finished: 79 passed, 5 skipped, 1 expected to fail, 3 exceptions, in 147.80 seconds ==========================
DO *NOT* COMMIT!
```

### PR DESCRIPTION
**Pull request by czgdp1807**
#### References to other Issues or PRs

Fix for #20292 on master

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by asmeurer** [comment]
You need to add the files that use this code to the external modules test list in bin/test_travis.sh. Otherwise we can't verify if the fix works from the Travis tests.

**Comment by asmeurer** [comment]
I think the fix I mentioned on the issue is better. Removing scipy would presumably break other things, since scipy is needed for some of the special functions, so presumably this would break things for any random variable whose PDF uses them. Also, 'scipy' is used in drv.py as well (though this particular issue doesn't come up in the tests for it).

**Comment by czgdp1807** [comment]
@asmeurer Please see if the changes are good.

**Comment by asmeurer** [comment]
It looks OK to me. Let's see if the tests pass. I don't know if you want to make this against the release branch instead of master, or if @oscarbenjamin will take care of that.

**Comment by czgdp1807** [comment]
Let me know I will edit the PR accordingly. The stats tests pass on my system.

**Comment by oscarbenjamin** [comment]
Thanks @asmeurer and @czgdp1807. This looks good to me. I can cherry-pick this for the release branch if the tests pass. Hopefully #20289 has fixed itself...

**Comment by oscarbenjamin** [comment]
> Hopefully #20289 has fixed itself...

Nope. I guess that's now the main release blocker!

**Comment by oscarbenjamin** [comment]
Okay, now this has actually failed in the tests:
```
========== tests finished: 3904 passed, 2 skipped, in 414.31 seconds ===========
+[[ '' == \t\r\u\e ]]
+[[ '' == \t\r\u\e ]]
+[[ '' == \t\r\u\e ]]
+[[ -n matchpy numpy scipy gmpy2 matplotlib theano llvmlite autowrap cython wurlitzer python-symengine=0.5.1 tensorflow numexpr ipython antlr-python-runtime>=4.7,<4.8 antlr>=4.7,<4.8 cloudpickle pyglet pycosat lfortran python-clang lxml ]]
+python
+cat
  File "<stdin>", line 49
    sympy/stats/,
                ^
SyntaxError: invalid syntax
```

File: bin/test_travis.sh [hunk 1]
```diff
@@ -187,6 +187,9 @@ test_list = [
     'sympy/logic',
     'sympy/assumptions',
 
+    #stats
+    sympy/stats/,
```
Comment (czgdp1807) [inline bin/test_travis.sh:191]:
```suggestion
    'sympy/stats/',
```

File: bin/test_travis.sh [hunk 2]
```diff
@@ -229,6 +232,9 @@ doctest_list = [
...
```
Comment (czgdp1807) [inline bin/test_travis.sh:236]:
```suggestion
    'sympy/stats',
```

**Pull request merged**